x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -396,6 +384,112 @@ public void testIsNullPushdownFilter() {
+            new OutOfRangeTestCase("double", "-1.0/0.0", "1.0/0.0"),
+            new OutOfRangeTestCase("scaled_float", "-1.0/0.0", "1.0/0.0")
+        );
+
+        for (OutOfRangeTestCase testCase : cases) {
+            List<String> trueForSingleValuesPredicates = List.of(
+                "<" + testCase.tooHigh,
+                "<=" + testCase.tooHigh,
+                ">" + testCase.tooLow,
+                ">=" + testCase.tooLow,
+                "!=" + testCase.tooHigh,
+                "!=" + testCase.tooLow,
+                "!= 0.0/0.0"
+            );
+            List<String> alwaysFalsePredicates = List.of(
+                "<" + testCase.tooLow,
+                "<=" + testCase.tooLow,
+                ">" + testCase.tooHigh,
+                ">=" + testCase.tooHigh,
+                "==" + testCase.tooHigh,
+                "==" + testCase.tooLow,
+                "< 0.0/0.0",
+                "<= 0.0/0.0",
+                "> 0.0/0.0",
+                ">= 0.0/0.0",
+                "== 0.0/0.0"
+            );
+
+            for (String truePredicate : trueForSingleValuesPredicates) {
+                String comparison = testCase.fieldName + truePredicate;
+                var query = "from test | where " + comparison;
+                Source expectedSource = new Source(1, 18, comparison);
+                String expectedLuceneQuery = LoggerMessageFormat.format(null, """
+                    {
+                      "esql_single_value" : {
+                        "field" : "{}",
+                        "next" : {
+                          "match_all" : {
+                            "boost" : 1.0
+                          }
+                        },
+                        "source" : "{}"
+                      }
+                    }""", testCase.fieldName, expectedSource);
+                doTestOutOfRangeFilterPushdown(query, expectedLuceneQuery);
+            }
+
+            for (String falsePredicate : alwaysFalsePredicates) {
+                String comparison = testCase.fieldName + falsePredicate;
+                var query = "from test | where " + comparison;
+                Source expectedSource = new Source(1, 18, comparison);
+                String expectedLuceneQuery = LoggerMessageFormat.format(null, """
+                    {
+                      "esql_single_value" : {
+                        "field" : "{}",
+                        "next" : {
+                          "bool" : {
+                            "must_not" : [
+                              {
+                                "match_all" : {
+                                  "boost" : 1.0
+                                }
+                              }
+                            ],
+                            "boost" : 1.0
+                          }
+                        },
+                        "source" : "{}"
+                      }
```

Comment:
String comparisons are finicky - next time walk the object tree.
check the overall query is QSV, then its wrapped query and that is either is match_all or a negated one.